src/parsers/csr_parser.cc

```diff
@@ -41,7 +41,12 @@ std::shared_ptr<Scene> CSRParser::parseCSR(std::string& filePath, RTCDevice devi
                     getNextLine(file, albedo);
                     materials[readStringProperty(materialId)] = std::make_shared<lambertian>(readXYZProperty(albedo));  
                 } else {
-                    materials[readStringProperty(materialId)] = std::make_shared<lambertian>(textures[texture_id]);
+                    std::shared_ptr<PixelImageTexture> plamb = std::dynamic_pointer_cast<PixelImageTexture>(textures[texture_id]);
+                    if (plamb) { // texture is a pixel lambert
```

Comment:
I can't pinpoint it, but the code smell seems off for this. I don't think it's a good idea to check instance by using dynamic pointers

Reply:
Sorry if I'm not understanding correctly, but if I am, then pixel Lambertian is similar to Lambertian but with a transparency field. High-key what we should do imo, is combine lambertian and pixel Lambertian, where our current Lambertian just is completely opaque. and then for materials in general, we should take in colour 4 in general, since being able to modify transparency seems like it should be a no brainer for any material type if we want to eventually create custom materials

This might require refactoring a chunk of the codebase, but I think it's the better play in the long run

Reply:
Wait so it's this current fix just a temp solution?